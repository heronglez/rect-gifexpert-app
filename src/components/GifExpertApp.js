import React, { useState } from 'react'
import { AddCategory } from './AddCategory'
import { GifGrid } from './GifGrid'

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Euphoria'])

    return (
        <>
            <h2>GiftExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr/>

            <ol>
                { 
                    categories.map(category => 
                        <GifGrid key={category} category={ category } />
                    )
                }
            </ol>
        </>
    )
}