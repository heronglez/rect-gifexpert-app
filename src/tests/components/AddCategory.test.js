import '@testing-library/jest-dom'

import { shallow } from 'enzyme';
import { AddCategory } from "../../components/AddCategory";

describe('Pruebas en <AddCategory />', () => { 

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories } />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories } />);
    })

    test('debe de mostrar correctamente ', () => { 

        expect( wrapper ).toMatchSnapshot();

     })

    test('debe de cambiar la caja de texto', () => {

        const input = wrapper.find('input')
        const value = 'Hola mundo'

        input.simulate('change', { target: { value }})

    })

    test('NO debe de postear la info onSubmit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect(setCategories).not.toHaveBeenCalled();

    })

    test('debe de llamar el setCategories y limpar la caja de texto', () => { 

        const value = 'doctor House'

        // 1. simular el inputChange
        wrapper.find('input').simulate('change', { target: { value } })
        
        // 2. simular submit
        wrapper.find('form').simulate('submit', { preventDefault(){} })

        // 3. setcategories se debe de haber llamado
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        

        // 4. inputValue debe de estar vacio.
        expect( wrapper.find('input').prop('value') ).toBe('')

     })
 })