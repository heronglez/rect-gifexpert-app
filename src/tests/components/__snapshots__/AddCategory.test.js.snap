// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrar correctamente  1`] = `ShallowWrapper {}`;
